src/module-elasticsuite-core/Index/IndexSettings.php

```diff
@@ -226,6 +226,74 @@ public function getDynamicIndexSettings($store)
         return $settings;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function getTimeBeforeGhost(): int
+    {
+        return $this->helper->getTimeBeforeGhost();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function parseIndexName(string $indexName): array
+    {
+        // 1. Remove prefix
+        $alias = $this->helper->getIndexAlias();
+        if (!str_starts_with($indexName, "${alias}_")) {
+            throw new \LogicException(sprintf('Index %s is not an elasticsuite index.', $indexName));
+        }
+        $indexName = str_replace("${alias}_", '', $indexName);
```

Comment:
I'm not so sure about throwing an exception nor the location of the method...

Reply:
Hum yes you're right, I'll update this to keep the original logic that return false for external index